x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/rest/RestSqlTestCase.java

```diff
@@ -1183,4 +1204,207 @@ public static void assertResponse(Map<String, Object> expected, Map<String, Obje
+            .keepOnCompletion(false);
+
+        String mode = randomMode();
+
+        Map<String, Object> expected = new HashMap<>();
+        expected.put(IS_PARTIAL_NAME, false);
+        expected.put(IS_RUNNING_NAME, false);
+        expected.put(COLUMNS_NAME, singletonList(columnInfo(mode, "1", "integer", JDBCType.INTEGER, 11)));
+        expected.put(ROWS_NAME, singletonList(singletonList(1)));
+        assertAsyncResponse(expected, runSql(builder, mode));
+    }
+
+    public void testAsyncTextWait() throws IOException {
+        RequestObjectBuilder builder = query("SELECT 1").waitForCompletionTimeout("1d").keepOnCompletion(false);
+
+        Map<String, String> contentMap = new HashMap<>() {
+            {
+                put("txt", "       1       \n---------------\n1              \n");
+                put("csv", "1\r\n1\r\n");
+                put("tsv", "1\n1\n");
+            }
+        };
+
+        for (String format : contentMap.keySet()) {
+            Response response = runSqlAsTextWithFormat(builder, format);
+
+            assertEquals(contentMap.get(format), responseBody(response));
+
+            assertTrue(hasText(response.getHeader(HEADER_NAME_ASYNC_ID)));
+            assertEquals("false", response.getHeader(HEADER_NAME_ASYNC_PARTIAL));
+            assertEquals("false", response.getHeader(HEADER_NAME_ASYNC_RUNNING));
+        }
+    }
+
+    public void testAsyncTextPaginated() throws IOException, InterruptedException {
+        final Map<String, String> acceptMap = new HashMap<>() {
+            {
+                put("txt", "text/plain");
+                put("csv", "text/csv");
+                put("tsv", "text/tab-separated-values");
+            }
+        };
+        final int fetchSize = randomIntBetween(1, 10);
+        final int fetchCount = randomIntBetween(1, 9);
+        bulkLoadTestData(fetchSize * fetchCount); // NB: product needs to stay below 100, for txt format tests
+
+        String format = randomFrom(acceptMap.keySet());
+        String mode = randomMode();
+        String cursor = null;
+        for (int i = 0; i <= fetchCount; i++) { // the last iteration (the equality in `<=`) checks on no-cursor & no-results
+            // start the query
+            RequestObjectBuilder builder = (hasText(cursor) ? cursor(cursor) : query("SELECT text, number FROM test")).fetchSize(fetchSize)
+                .waitForCompletionTimeout("0d") // don't wait at all
+                .keepOnCompletion(true)
+                .keepAlive("1d") // keep "forever"
+                .mode(mode)
+                .binaryFormat(false); // prevent JDBC mode to (ignore `format` and) enforce CBOR
+            Response response = runSqlAsTextWithFormat(builder, format);
+
+            Character csvDelimiter = ',';
+
+            assertEquals(200, response.getStatusLine().getStatusCode());
+            assertEquals(response.getHeader(HEADER_NAME_ASYNC_PARTIAL), response.getHeader(HEADER_NAME_ASYNC_RUNNING));
```

Comment:
> assertEquals is about comparing a (actual) value to something that's expected. 

Not sure, I might have taken editor's suggestion to "simplify" the code. The documentation is not "contractual" enough, IMO, but parameter naming might support your case. :-) So I see your point and I've refactored to `assertTrue()` (tho maybe a bit less easy to read).

> Just few lines above you used the right way for these assertions

This case differs from above since both "true" or "false" are valid, as long as both headers have the same value.